api/v1alpha1/argocd_conversion.go

```diff
@@ -0,0 +1,577 @@
+package v1alpha1
+
+import (
+	"reflect"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	"github.com/argoproj-labs/argocd-operator/api/v1beta1"
+)
+
+var conversionLogger = ctrl.Log.WithName("conversion-webhook")
+
+// ConvertTo converts this (v1alpha1) ArgoCD to the Hub version (v1beta1).
+func (src *ArgoCD) ConvertTo(dstRaw conversion.Hub) error {
+	conversionLogger.Info("v1alpha1 to v1beta1 conversion requested.")
+	dst := dstRaw.(*v1beta1.ArgoCD)
+
+	// ObjectMeta conversion
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Spec conversion
+
+	// sso field
+	sso := ConvertAlphaToBetaSSO(src.Spec.SSO)
+
+	// in case of conflict, deprecated fields will have more priority during conversion to beta
```

Comment:
deprecated fields will have more priority ?

Reply:
Here, the idea of conversion webhook is for smooth migration of existing v1alpha1 users to v1beta1. If any existing users are using deprecated fields that means they have not updated their sso configurations to use new keys hence having more priority to deprecated fields in such cases makes more sense to me.

Reply:
I think its fine either way, the current checks prevent a confict like that from happening in the first place 
i.e, they can't have the same fields with conflicting info in `.spec.dex` and `.spec.sso.dex` in their `v1alpha1` CRs anyway